src/routes/Profile.jsx

```diff
@@ -74,85 +86,143 @@ const Profile = () => {
   }, []);
 
   return (
-    <Box sx={{ display: 'flex', flexDirection: 'column', gap: '20px', padding: '30px' }}>
-      <Box sx={{ display: 'flex', flexDirection: 'column', gap: '20px' }}>
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        gap: '20px',
+        padding: '30px'
+      }}
+    >
+      <Box sx={{ display: 'flex', flexDirection: 'column', alignItems: 'center', gap: '20px' }}>
         <Typography sx={{ fontWeight: 'bold', fontSize: '18px' }}>Profile Information</Typography>
 
         <Typography>
           User WebId:{' '}
-          <Link href={session.info.webId} target="_blank" rel="noreferrer">
-            {session.info.webId}
+          <Link
+            href={user === 'personal' ? session.info.webId : selectedUser.webId}
+            target="_blank"
+            rel="noreferrer"
+          >
+            {user === 'personal' ? session.info.webId : selectedUser.webId}
           </Link>
         </Typography>
 
         {/* TODO: Refactor/optimize the form below once we have more input */}
         {/* fields to update profile for */}
-        <Box style={{ display: 'flex', flexDirection: 'column', gap: '15px' }}>
-          <form onSubmit={handleUpdateProfile}>
-            <Box
-              sx={{
+        {user === 'personal' ? (
```

Comment:
This "personal" vs "public" is section is pretty large. Could we factor them out into separate components?

Reply:
Yeah, I could do that. I'll see what I could do to get these elements separated.

Reply:
Got this separated.